org.eclipse.jdt.debug.tests/tests/org/eclipse/jdt/debug/tests/core/ExecutionEnvironmentTests.java

```diff
@@ -259,4 +264,18 @@ public void testEEHomeVariableInvalidArgument() throws Exception {
 		}
 		assertNotNull("Test should have thrown an exception", null);
 	}
+
+	public void testSerializability() throws Exception {
+		IExecutionEnvironment ee = JavaRuntime.getExecutionEnvironmentsManager().getEnvironment("JavaSE-21");
```

Comment:
since you want to serialize IExecutionEnvironment shouldn't that extends Serializable publicly?

Reply:
Sure.
Since `IExecutionEnvironment` is marked as `@noimplement/@noextend` there shouldn't be custom implementations anyways or at least custom impls are aware that they are off the regular paths.